core/datastore/src/main/java/com/google/samples/apps/nowinandroid/core/datastore/NiaPreferencesDataSource.kt

```diff
@@ -33,54 +31,85 @@ class NiaPreferencesDataSource @Inject constructor(
     val userDataStream = userPreferences.data
         .map {
             UserData(
-                bookmarkedNewsResources = it.bookmarkedNewsResourceIdsList.toSet(),
-                followedTopics = it.followedTopicIdsList.toSet(),
-                followedAuthors = it.followedAuthorIdsList.toSet(),
+                bookmarkedNewsResources = it.bookmarkedNewsResourceIdsMap.keys,
```

Comment:
Won't this take all the keys? Should it be filtering on the values as well?

Reply:
Yes, it'll take all the keys. No, it shouldn't filter on the values because the value irrelevant (it's always `true`). The presence of a key in the map indicates that it's a bookmark. 

From `user_preferences.proto`: 

```
// Each map is used to store a set of string IDs. The bool has no meaning, but proto3 doesn't
    // have a Set type so this is the closest we can get to a Set.
    map<string, bool> followed_topic_ids = 13;
    map<string, bool> followed_author_ids = 14;
    map<string, bool> bookmarked_news_resource_ids = 15;
```

I chose `bool` as the value type because it has the smallest size. 

Should I document this elsewhere as well? If it wasn't obvious to you, it's likely not obvious to others.

Reply:
Gotcha!